x-pack/solutions/security/plugins/security_solution/server/lib/telemetry/tasks/custom_response_actions_rule.ts

```diff
@@ -0,0 +1,142 @@
+      const log = newTelemetryLogger(logger.get('response_actions_rules'), mdc);
+      const usageCollector = sender.getTelemetryUsageCluster();
+      const usageLabelEndpointPrefix: string[] = [
+        'security_telemetry',
+        'endpoint-response-actions-rules',
+      ];
+      const usageLabelOsqueryPrefix: string[] = [
+        'security_telemetry',
+        'osquery-response-actions-rules',
+      ];
+      const trace = taskMetricsService.start(taskType);
+
+      log.l('Running response actions rules telemetry task');
+
+      try {
+        const [clusterInfoPromise, licenseInfoPromise] = await Promise.allSettled([
+          receiver.fetchClusterInfo(),
+          receiver.fetchLicenseInfo(),
+        ]);
+
+        const clusterInfo = safeValue(clusterInfoPromise);
+        const licenseInfo = safeValue(licenseInfoPromise);
+
+        const {
+          body: { aggregations },
+        } = (await receiver.fetchResponseActionsRules()) ?? {};
+
+        if (!aggregations || !aggregations.actionTypes) {
+          log.debug('no custom response action rules found');
+          await taskMetricsService.end(trace);
+          return 0;
+        }
+
+        const responseActionRules = (
+          aggregations as unknown as ResponseActionsRuleResponseAggregations
+        ).actionTypes.buckets.reduce<ResponseActionRules>((acc, agg) => {
+          if (agg.key === '.endpoint') {
+            acc.endpoint = agg.rulesInfo.buckets.map((rule) => rule.key);
+          } else if (agg.key === '.osquery') {
+            acc.osquery = agg.rulesInfo.buckets.map((rule) => rule.key);
+          }
+          return acc;
+        }, {} as ResponseActionRules);
+
+        const shouldNotProcessTelemetry =
+          responseActionRules.endpoint === undefined ||
+          responseActionRules.osquery === undefined ||
+          responseActionRules.endpoint.length === 0 ||
+          responseActionRules.osquery.length === 0;
+
+        if (shouldNotProcessTelemetry) {
+          return 0;
+        }
+
+        const responseActionsRulesTelemetryData = responseActionsCustomRuleTelemetryData(
+          responseActionRules,
+          clusterInfo,
+          licenseInfo
+        );
+
+        log.l('Custom response actions rules data', {
+          data: JSON.stringify(responseActionsRulesTelemetryData),
+        });
+
+        usageCollector?.incrementCounter({
+          counterName: createUsageCounterLabel(usageLabelEndpointPrefix),
+          counterType: 'response_actions_rules_count',
+          incrementBy: responseActionsRulesTelemetryData.response_actions.endpoint.count,
+        });
+
+        usageCollector?.incrementCounter({
+          counterName: createUsageCounterLabel(usageLabelOsqueryPrefix),
+          counterType: 'response_actions_rules_count',
+          incrementBy: responseActionsRulesTelemetryData.response_actions.osquery.count,
+        });
+
+        const batches = batchTelemetryRecords(
```

Comment:
you can use the AsyncSender, which takes care of splitting the record into batches and also improves the error handling and retries. Example [here](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/server/lib/telemetry/tasks/endpoint.ts#L88-L96).

Reply:
done 2cd4c87c9694e445b6dcbd0e6cb4c4099677dfb7